### PR TITLE
Add HCP Packer registry image metadata to builder artifacts

### PR DESCRIPTION
**Pull request by nywilken**

### CODE REVIEW

File: builder/azure/arm/artifact.go [hunk 1]
```diff
@@ -249,13 +250,45 @@ func (*Artifact) Destroy() error {
...
+	if a.isManagedImage() {
+		id = a.ManagedImageId
+		location = a.ManagedImageLocation
+
+		labels := make(map[string]interface{})
+		labels["os_type"] = a.OSType
+		labels["managed_image_resourcegroup_name"] = a.ManagedImageResourceGroupName
+		labels["managed_image_name"] = a.ManagedImageName
+		if a.ManagedImageSharedImageGalleryId != "" {
+			labels["managed_image_sharedimagegallery_id"] = a.ManagedImageSharedImageGalleryId
```
Comment (SwampDragons) [inline builder/azure/arm/artifact.go:265]: We probably also need the SIG name, as well as the SIG Image Name, and the SIG version.

**Review by SwampDragons — APPROVED** [review]
Super solid. We need a few more labels for images that are associated with a SIG, noted inline. Otherwise, this works as expected! (Check the vendoring; I had to change the SDK versioning to v0.2.5 to get it to build.

**Review by azr — APPROVED** [review]
LGTM

File: builder/azure/arm/artifact.go (hunk 1 above)
Comment (nywilken) [reply builder/azure/arm/artifact.go:265]: See PR #150

**Pull request merged**